File: shoe.py
class Shoe:
    def __init__(self, brand, model, new_price, old_price, all_sizes, available_sizes, provider, provider_url, images):
        self.brand = brand
        self.model = model
        self.new_price = new_price
        self.old_price = old_price
        self.all_sizes = all_sizes
        self.available_sizes = available_sizes
        self.provider = provider
        self.provider_url = provider_url
        self.images = images

    def __str__(self):
        return (
            f'Brand: {self.brand}\n'
            f'Model: {self.model}\n'
            f'New price: {self.new_price}\n'
            f'Old price: {self.old_price}\n'
            f'Sizes: {self.all_sizes}\n'
            f'Available sizes: {self.available_sizes}\n'
            f'Provider: {self.provider}\n'
            f'Provider url: {self.provider_url}\n'
            f'Images: {self.images}\n'
        )

    def to_dict(self):
        return {
            'brand': self.brand,
            'model': self.model,
            'new_price': self.new_price,
            'old_price': self.old_price,
            'sizes': self.all_sizes,
            'available_sizes': self.available_sizes,
            'provider': self.provider,
            'provider_url': self.provider_url,
            'images': self.images
        }

File: src/main/java/com/example/footrade/entity/Shoe.java
package com.example.footrade.entity;

import lombok.Data;
import org.bson.types.ObjectId;
import org.springframework.data.annotation.Id;
import org.springframework.data.mongodb.core.mapping.Document;
import org.springframework.data.mongodb.core.mapping.Field;

import java.math.BigDecimal;
import java.util.Date;
import java.util.List;

@Document(collection = "shoes")
@Data
public class Shoe {
    @Id
    private ObjectId id;
    private String brand;
    private String model;
    @Field("new_price")
    private BigDecimal newPrice;
    @Field("old_price")
    private BigDecimal oldPrice;
    private List<String> sizes;
    @Field("available_sizes")
    private List<String> availableSizes;
    private String provider;
    private String url;
    private List<String> images;
    @Field("initial_image")
    private String initialImage;
    @Field("on_sale")
    private Boolean onSale;
    @Field("last_updated")
    private Date lastUpdated;
}
